sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinals.scala

```diff
@@ -27,13 +27,23 @@ import org.apache.spark.sql.types.IntegerType
  * Replaces ordinal in 'order by' or 'group by' with UnresolvedOrdinal expression.
  */
 object SubstituteUnresolvedOrdinals extends Rule[LogicalPlan] {
-  private def isIntLiteral(e: Expression) = e match {
+  private def containIntLiteral(e: Expression): Boolean = e match {
     case Literal(_, IntegerType) => true
+    case Cube(_, groupByExprs) => groupByExprs.exists(containIntLiteral)
+    case Rollup(_, groupByExprs) => groupByExprs.exists(containIntLiteral)
+    case GroupingSets(_, flatGroupingSets, groupByExprs) =>
+      flatGroupingSets.exists(containIntLiteral) || groupByExprs.exists(containIntLiteral)
     case _ => false
   }
 
+  private def resolveOrdinal(expression: Expression): Expression = expression match {
```

Comment:
the name is weird as it returns `UnresolvedOrdinal`. how about `substituteUnresolvedOrdinal`?

Reply:
> the name is weird as it returns `UnresolvedOrdinal`. how about `substituteUnresolvedOrdinal`?

Done and updated some missed logic.